sources/identify.php

```diff
@@ -1205,157 +1205,217 @@ function authenticateThroughAD(string $username, array $userInfo, string $passwo
+ */
+function authenticateUser(string $username, string $passwordClear, array $ldapHandler, array $SETTINGS, Language $lang): array
+{
     try {
-        // 2- Get user info from AD
-        // We want to isolate attribute ldap_user_attribute or mostly samAccountName
-        $userADInfos = $ldapConnection->query()
-            ->where((isset($SETTINGS['ldap_user_attribute']) ===true && empty($SETTINGS['ldap_user_attribute']) === false) ? $SETTINGS['ldap_user_attribute'] : 'samaccountname', '=', $username)
+        $userAttribute = $SETTINGS['ldap_user_attribute'] ?? 'samaccountname';
+        $userADInfos = $ldapHandler['connection']->query()
+            ->where($userAttribute, '=', $username)
             ->firstOrFail();
-
-        // Is user enabled? Only ActiveDirectory
-        if ($SETTINGS['ldap_type'] === 'ActiveDirectory' && isset($activeDirectoryExtra) === true && $activeDirectoryExtra instanceof ActiveDirectoryExtra) {
-            if ($activeDirectoryExtra->userIsEnabled((string) $userADInfos['dn'], $ldapConnection) === false) {
-                return [
-                    'error' => true,
-                    'message' => "Error : User is not enabled",
-                ];
-            }
+        
+        // Verify user status for ActiveDirectory
+        if ($ldapHandler['type'] === 'ActiveDirectory' && !$ldapHandler['handler']->userIsEnabled((string) $userADInfos['dn'], $ldapHandler['connection'])) {
+            return [
+                'error' => true,
+                'message' => "Error: User is not enabled"
+            ];
         }
-
-        // 3- User auth attempt
-        // For AD, we use attribute userPrincipalName
-        // For OpenLDAP and others, we use attribute dn
-        $userAuthAttempt = $ldapConnection->auth()->attempt(
-            $SETTINGS['ldap_type'] === 'ActiveDirectory' ?
-                $userADInfos['userprincipalname'][0] :  // refering to https://ldaprecord.com/docs/core/v2/authentication#basic-authentication
-                $userADInfos['dn'],
-            $passwordClear
-        );
-
-        // User is not auth then return error
-        if ($userAuthAttempt === false) {
+        
+        // Attempt authentication
+        $authIdentifier = $ldapHandler['type'] === 'ActiveDirectory' 
+            ? $userADInfos['userprincipalname'][0] 
+            : $userADInfos['dn'];
+            
+        if (!$ldapHandler['connection']->auth()->attempt($authIdentifier, $passwordClear)) {
             return [
                 'error' => true,
-                'message' => "Error: User is not authenticated",
+                'message' => "Error: User is not authenticated"
             ];
         }
-
-    } catch (\LdapRecord\Query\ObjectNotFoundException $e) {
-        $error = $e->getDetailedError();
+        
         return [
-            'error' => true,
-            'message' => $lang->get('error_bad_credentials'),
+            'error' => false,
+            'user_info' => $userADInfos
```

Comment:
This is not necessarily a problem but I wonder what exactly `$userADInfos` contains.

Reply:
If I remember well, $userADInfos contains a set of LDAP user attributes such as DN, mail, givenname, etc.